### ISSUE TITLE
12-hour time format breaks filters

### ISSUE THREAD

**Issue by AtomicFS** [engelsystem/engelsystem#1490]
For some reason my system gets detected as using 12-hour time format (and also `mm/dd/yyyy`) without the option to change this to 24-hour format (and `dd-mm-yyyy` or equivalent).

The problem with 12-hour format is that there is no option for `AM` and `PM`, meaning that I cannot filter for anything after mid-day (noon).

Personally, I would just discard the 12-hour auto-detected option and enforce 24-hour time format.

**Comment by Enterprize1** [comment]
The engelsystem uses the browser's default time input, which automatically detects the time format based on the regional settings. Overriding this behavior would require replacing the built-in time input with a custom implementation.

For me it looks like this on chrome with US region settings:

![Image](https://github.com/user-attachments/assets/0dca24c8-1d38-474a-ba77-adcb5481309c)

I would recommend keeping the default behavior. The time format used by the browser is consistent for all websites. If the time format is not as expected, adjusting the browser's region settings would be a simpler solution.

**Comment by AtomicFS** [comment]
Yeah, there lies the problem, in the Language settings - I want English, but not 12-hour time format.

**Comment by teal-bauer** [comment]
@AtomicFS Your initial message said you cannot choose AM/PM in 12h format, is that the case? Because the UI should come from your browser, this would be quite unexpected as an unfixed bug. Or is it just that you want to be able to choose the datetime locale separately from your system/browser locale?

**Comment by AtomicFS** [comment]
Hey, sorry for the late reply.

> Your initial message said you cannot choose AM/PM in 12h format, is that the case?

Yes, that was the case. The filters were not usable when in 12-hour format.

> Or is it just that you want to be able to choose the datetime locale separately from your system/browser locale?

I would actually like that - to have the option to override this setting. I hate 12-hour time format, and I don't even know why the engelsystem decided to use 12-hour format instead of 24-hour as defined in my system. Probably some browser nonsense.
